src/index.ts

```diff
@@ -437,22 +446,41 @@ export async function init(config: IClientConfig): Promise<EppoClient> {
           e,
         );
         initFromConfigStoreError = e;
+        return 'did not finish';
       });
-    const attemptInitFromFetch = instance
+
+    const attemptInitFromFetch: RacingPromise = instance
       .fetchFlagConfigurations()
-      .then(() => 'fetch')
+      .then(() => 'fetch' as ConfigSources)
       .catch((e) => {
         applicationLogger.warn('Eppo SDK encountered an error initializing from fetching', e);
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         initFromFetchError = e;
+        return 'did not finish';
       });
 
+    const persistentStoreHeadStartMs =
+      config.localStoreHeadstartMs ?? DEFAULT_LOCAL_STORE_HEAD_START_MS;
+    const headStart = new Promise<ConfigSources>((resolve) => {
+      setTimeout(() => resolve('head start'), persistentStoreHeadStartMs);
+    });
+
+    const headStartWinner = await Promise.race([headStart, attemptInitFromConfigStore]);
```

Comment:
~I reckon we could still call this "initializationSource" no need to bring headStartIntoThis~

OK I see what's happening, I think we'll want more code comments because it's a bit confusing.

Basically we want the config store to race against a 10ms timeout, but if the timeout wins, we allow it to keep going in a new race against fetch. If it wins, we don't bother with fetch at all.